### DISCUSSION TITLE
AWS/EC2  Build from source help

### DISCUSSION THREAD

**Discussion by cottlestone** [ipxe/ipxe#1132] [Q&A] [answered]
Hello all, 

I'm trying to build an AMI from source for one of our engineers since they don't have rights on our AWS environment in us-gov-west.
I'm having trouble understanding how the aws-import is capturing the ipxe.usb build that I made earlier with the corresponding MAKE command. (This is my first time setting up iPXE any help is appreciated) 

I am under the impression that the aws-import script creates an AMI out of a snapshot of the instance I just made the build on. But what's throwing me for a loop are the arguments in the aws-import script. There's the optional image name , which I'm guessing is the what the new AMI will be called. And then there's the image ? Image of what ? and for what ? 

When  creating an AMI , I don't believe we have the ability to choose an ID so what is this image parameter asking for ?

**Comment by NiKiZe** [comment]
Git commit messages often has some information, so see https://github.com/ipxe/ipxe/commit/d16535aa4fcb30c78559103e1bc994fa99bf5748

Other commits for the script https://github.com/ipxe/ipxe/commits/de8a0821c7bc737e724fa3dfb6d89dc36f591d7a/contrib/cloud/aws-import

**Comment by mcb30** [comment, marked as answer]
@cottlestone The `image` parameter is the iPXE disk image that you have built.  For example, if you want to create an AMI for x86 BIOS boot (the most common setup), try:

```
make CONFIG=cloud EMBED=config/cloud/aws.ipxe bin/ipxe.usb
../contrib/cloud/aws-import -r us-gov-west bin/ipxe.usb
```

**Reply by cottlestone** [reply]
Thanks for clarifying this ! I understand now , the import is building the ami from the .usb file by reading block by block. I'm going to run the aws-import once ive added the right permissions. Thanks again !
